### PR TITLE
JK - Test Technique 28/09/2023 Joan KUCUKOGLU

### PR DESCRIPTION
**Pull request by JoanKUCUKOGLU**
Exercice réalisé sans test (toujours lié aux soucis de SSL). Vous pouvez ainsi étudier ma façon d'aborder l'exercice.

Bonne journée à vous.
Joan.

### CODE REVIEW

File: back/src/controller/router.ts [hunk 1]
```diff
@@ -1,74 +1,96 @@
...
     }
   });
 
-  router.post('/transfer', async (req, res) => {
+  router.post("/transfer", async (req, res) => {
     try {
       // TODO: implement a crypto transfer here
-      const hash = "0x1234567890123456789012345678901234567890123456789012345678901234"
-      res.json({hash})
+      const request: TransferRequest = req.body;
```
Comment (guenoledc) [inline back/src/controller/router.ts:34]: Initializing a class that has a constructor with a simple assignment seems wrong. Also, the req.body will not hold a bigint so conversion is needed

File: front/src/components/LandingPage.vue [hunk 2]
```diff
@@ -1,63 +1,81 @@
...
-  console.log("wallets", wallets.value);
+  wallets.value = await listWallets()
+  console.log('wallets', wallets.value)
+}
+
+async function proceedTransfer() {
+  lastTransactionHash.value = await transferToken(
+    fromAddress.value,
+    toAddress.value,
+    BigInt(amount.value)
```
Comment (guenoledc) [inline front/src/components/LandingPage.vue:44]: Are you sure the BigInt will be json serialized in the http requests ?

**Comment by guenoledc** [comment]
@JoanKUCUKOGLU thanks for the sharing

This is a good work. Congrats.

I have put a few comments on your files.

I'd like also to get to the bottom of your ssl config. Can you try to install curl and run the following command to display the ssl certificate.

```sh
curl -v  https://cacib-saturn-test.francecentral.cloudapp.azure.com
```
It works well on my side and it displays the following result
```
*   Trying 20.199.115.230:443...
* Connected to cacib-saturn-test.francecentral.cloudapp.azure.com (20.199.115.230) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: CN=cacib-saturn-test.francecentral.cloudapp.azure.com
*  start date: Aug 16 01:39:20 2023 GMT
*  expire date: Nov 14 01:39:19 2023 GMT
*  subjectAltName: host "cacib-saturn-test.francecentral.cloudapp.azure.com" matched cert's "cacib-saturn-test.francecentral.cloudapp.azure.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* using HTTP/1.1
> GET / HTTP/1.1
> Host: cacib-saturn-test.francecentral.cloudapp.azure.com
> User-Agent: curl/8.1.2
> Accept: */*
> 
< HTTP/1.1 200 OK
< Server: nginx/1.18.0 (Ubuntu)
< Date: Fri, 29 Sep 2023 09:43:49 GMT
< Content-Length: 0
< Connection: keep-alive
< Vary: Origin
< 
* Connection #0 to host cacib-saturn-test.francecentral.cloudapp.azure.com left intact
```

Also, try running the backend inside a docker instance (ubuntu for instance) as I feel that your PC is lacking the root certificate for the issuer R3 of let's encrypt

**Comment by JoanKUCUKOGLU** [comment]
@guenoledc i just tried the command that you told me to try and and got this result :

```
COMMENTAIRES : GET https://cacib-saturn-test.francecentral.cloudapp.azure.com/ with 0-byte payload
COMMENTAIRES : received 0-byte response of content type 

StatusCode        : 200
StatusDescription : OK
Content           : {}
RawContent        : HTTP/1.1 200 OK
                    Connection: keep-alive
                    Vary: Origin
                    Content-Length: 0
                    Date: Fri, 29 Sep 2023 13:56:21 GMT
                    Server: nginx/1.18.0 (Ubuntu)


Headers           : {[Connection, keep-alive], [Vary, Origin], [Content-Length, 0], [Date, Fri, 29 Sep 2023 13:56:21 GMT]...}
RawContentLength  : 0
```

**Comment by guenoledc** [comment]
So it did not fail with curl. 

However it does not display the validation (or not) of the server certificate. 

Try running in docker !

**Comment by guenoledc** [comment]
Thanks for the work done. We will not merge this PR so let's close it.
You can continue to update it if needed

**Comment by JoanKUCUKOGLU** [comment]
@guenoledc Hello,
Sorry for this late message, I had to work with different environments and networks to try and find why we had theses bugs about SSL. After a few tries, I think this might be related to our intern network and restrictions so it might not be because of your project or configuration.

I hope this will help you on your futures test. 

Have a great day.

**Pull request closed without merging**